Add naive path filtering

orgrinrt opened the pull request:
## What 

This plugin seems nifty, but to make it fit into my workflow, I needed to be able to filter certain paths to hide any found todos from the item tree. So I made very rudimentary path filtering for my own use.

I thought I'd see if it'd be of any use here. Not sure if this is something that should be integrated into the plugin itself, but I find it's often good to just see if it could be salvaged.

(demo gif at the end for tldr)

## Why

For example, having addons in the project often results in a very bloated TODO view with a lot of unrelated entries.

## How

The implementation is very naive, so it's just an array of strings that are matched with `string.begins_with()` to see if it should be displayed in the tree. So for example, having a value of `addons, scenes/something, assets/two/levels` in the input field would result in anything within `addons` dir to not display, as well as the other two cases. 

There's a potential source of annoyance due to the use of `begins_with`: If the user has folders `scene` and `scenes`, writing down `scene` would also filter out `scenes`. Or if the user has a file `addons_list.txt` in the project root and filtering rule of `addons`, the same happens. To work around this, I found it's best to end directory paths with a trailing `/` but it's not really ideal.

This filtering could also probably do a better job if it was applied earlier in the process, namely when the `todo_items` are gathered. Now it just skips any matches in the tree building part.

You'd also probably want to write this better using regex or something in that vein, but this does the trick for me, and I thought it might be of use. If nothing else, maybe this gives an idea of how to implement this behaviour in a completely different manner, if the idea seems useful for the plugin. The filtering rules, for example, could be written in the same UX as the actual item filtering part of settings and make use of the same components.

![todo-manager-godot-pr-demo](https://user-images.githubusercontent.com/23649183/101417127-d9f73f80-38f3-11eb-922b-e7629049e336.gif)

OrigamiDev-Pete commented:
This seems like a great addition! I'll review your code shortly.

OrigamiDev-Pete commented on addons/Todo_Manager/Dock.gd:
```diff
@@ -41,6 +42,11 @@ func build_tree() -> void:
+		for ignore_path in ignore_paths:
+			if todo_item.script_path.begins_with("res:///" + ignore_path):
+				ignore = true;
+		if ignore: continue;
```
I'm not a huge fan of one-liners like this. Could it be changed to:
```
if ignore:
    continue
```

orgrinrt replied:
Fixed and pushed the change

OrigamiDev-Pete reviewed (changes requested):
I think it looks pretty good. I have just one little nit-pick but otherwise I'm happy to merge it. Thanks for your contribution!

OrigamiDev-Pete commented:
I was just trying out your latest commit and realised that for some reason your project directory starts with "res:///" and mine starts with "res://". I'm not entirely sure why that is so I just added a case for both and it works. I also changed some minor styling things. Let me know if it still works for you.

orgrinrt commented:
> I was just trying out your latest commit and realised that for some reason your project directory starts with "res:///" and mine starts with "res://". I'm not entirely sure why that is so I just added a case for both and it works.

Ah, didn't think of that. Could it be a difference with environments? Anyway good catch, I didn't really test this elsewhere. 

> I also changed some minor styling things. 

Yeah, I rarely write gdscript so I got the style wrong there. My bad.

> Let me know if it still works for you.

As long as it works and is something that the plugin benefits from, then great! Always happy to have helped, even if a minor thing like this 🙂
